tests/expressions_case/tests.py

```diff
@@ -1243,3 +1243,31 @@ def test_conditional_aggregation_example(self):
             ),
             {'regular': 2, 'gold': 1, 'platinum': 3}
         )
+
+    def test_annotation_with_values_and_when(self):
```

Comment:
I think we could include a more descriptive test name instead of a docstring, e.g. `test_annotate_order_by_values_annotation_not_in_order_by`. Also, these tests are in `CaseDocumentationExamples` which is a bit confusing since these aren't documentation examples. Any reason to use something like [the simpler test](https://code.djangoproject.com/attachment/ticket/25316/25312-test.diff) that I attached to the ticket?